### PR TITLE
Exclude test files from crate

### PR DESCRIPTION
**Pull request by sisuresh**
### Why

Make the crate smaller (context - https://github.com/stellar/rs-soroban-env/issues/1294#issuecomment-1845729881)

### CODE REVIEW

File: tests/add_i128/Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,8 @@ edition = "2021"
 publish = false
 rust-version = "1.74"
 
+exclude = ["test_snapshots/"]
```
Comment (leighmcculloch) [inline tests/add_i128/Cargo.toml:10]: The examples aren't published so it isn't necessary to add it here.

File: soroban-sdk/Cargo.toml [hunk 2]
```diff
@@ -10,6 +10,8 @@ version.workspace = true
 edition = "2021"
 rust-version = "1.74"
 
+exclude = ["test_snapshots/", "tests/"]
```
Comment (leighmcculloch) [inline soroban-sdk/Cargo.toml:13]: The `sprban-sdk/tests` directory doesn't exist. What are you wanting to exclude with `tests/`?

**Review by leighmcculloch — COMMENTED** [review]
Thanks for fixing this. Couple inline comments.

File: soroban-sdk/Cargo.toml (hunk 2 above)
Comment (sisuresh) [reply soroban-sdk/Cargo.toml:13]: Oh interesting. This was meant to exclude src/tests, and it seems to do that when I check `cargo package --list`. I'll update the path anyways.

File: tests/add_i128/Cargo.toml (hunk 1 above)
Comment (sisuresh) [reply tests/add_i128/Cargo.toml:10]: Ah I was going off of `cargo package --list`, but I see the `publish = false` in the toml files. Fixed.

**Review by leighmcculloch — APPROVED** [review]

**Pull request merged**